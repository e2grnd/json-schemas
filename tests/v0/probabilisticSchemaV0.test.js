const fs = require('fs');
const path = require('path');
const Ajv = require('ajv/dist/2020').default;
const ajv = new Ajv({ strict: false, allErrors: true});

const probabilisticSchema = require('../../v0/input.probabilistic.schema.json');
const validate_probabilistic = ajv.compile(probabilisticSchema);


describe('Probabilistic data validation', () => {
  it('validates passing data against the schema', () => {
    const passingData = require('../../data/v0/passing_probabilistic_v0.json');
    const isValid = validate_probabilistic(passingData);
    expect(validate_probabilistic.errors).toBeNull();
    expect(isValid).toBeTruthy(); // Assert that the data passes validation
  });

  it('validates failing data against the schema', () => {
    const failingData = require('../../data/v0/failing_probabilistic_v0.json');
    const isValid = validate_probabilistic(failingData);
    expect(validate_probabilistic.errors).toMatchSnapshot(); // Assert that the data fails validation
  });
})
