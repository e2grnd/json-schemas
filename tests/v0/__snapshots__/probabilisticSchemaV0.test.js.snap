// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probabilistic data validation validates failing data against the schema 1`] = `
[
  {
    "instancePath": "/type",
    "keyword": "enum",
    "message": "must be equal to one of the allowed values",
    "params": {
      "allowedValues": [
        "discrete",
        "continuous",
      ],
    },
    "schemaPath": "#/properties/type/enum",
  },
  {
    "instancePath": "/states/levels/17",
    "keyword": "type",
    "message": "must be number",
    "params": {
      "type": "number",
    },
    "schemaPath": "#/properties/states/properties/levels/items/type",
  },
  {
    "instancePath": "/standardDeviation",
    "keyword": "type",
    "message": "must be number,null",
    "params": {
      "type": [
        "number",
        "null",
      ],
    },
    "schemaPath": "#/properties/standardDeviation/type",
  },
]
`;
